### ISSUE TITLE
Failing on compilation due to `TS2403: Subsequent variable declarations must have the same type.`

### ISSUE THREAD

**Issue by NickGlenn** [rollup/plugins#204]
- Rollup Plugin Name: typescript
- Rollup Plugin Version: 3.0.0
- Rollup Version: 1.30.1, 1.27.14
- Operating System (or Browser): Windows, OSX
- Node Version: 8.10.0 (Win), 12.15.0 (Win), 12.15.0 (OSX)

### How Do We Reproduce?

[Repl.it Project Replicating the Issue](https://repl.it/repls/PeskyBigheartedForms)

I've seen this occur on a couple different libraries know, but the most notable one is `react-router-dom`. Set up a React project and install `react-router-dom` along with the types, `@types/react-router-dom`.

### Expected Behavior

Ideally it should compile unless there's actual user error. Using the exact same project but with the old [rollup-plugin-typescript](https://github.com/rollup/rollup-plugin-typescript) module works flawlessly.

### Actual Behavior

Build fails due to `TS2403: Subsequent variable declarations must have the same type.  Variable 'History' ...`.

```
src/index.tsx → dist/bundle.js...
[!] (plugin typescript) Error: @rollup/plugin-typescript TS2403: Subsequent variable declarations must have the same type.  Variable 'History' must be of type 'typeof import("/home/runner/PeskyBigheartedForms/node_modules/@types/history/index")', but here has type '{ new (): import("/home/runner/PeskyBigheartedForms/node_modules/@types/history/index"); prototype: import("/home/runner/PeskyBigheartedForms/node_modules/@types/history/index"); }'.
node_modules/typescript/lib/lib.dom.d.ts (9211:13)

9211 declare var History: {
                 ~~~~~~~

  node_modules/@types/history/index.d.ts:6:1
    6 export as namespace History;
      ~~~~~~
    'History' was also declared here.

Error: @rollup/plugin-typescript TS2403: Subsequent variable declarations must have thesame type.  Variable 'History' must be of type 'typeof import("/home/runner/PeskyBigheartedForms/node_modules/@types/history/index")', but here has type '{ new (): import("/home/runner/PeskyBigheartedForms/node_modules/@types/history/index"); prototype: import("/home/runner/PeskyBigheartedForms/node_modules/@types/history/index"); }'.
    at error (/home/runner/PeskyBigheartedForms/node_modules/rollup/dist/rollup.js:5365:30)
    at throwPluginError (/home/runner/PeskyBigheartedForms/node_modules/rollup/dist/rollup.js:11725:12)
    at Object.error (/home/runner/PeskyBigheartedForms/node_modules/rollup/dist/rollup.js:12694:24)
    at diagnostics.filter.forEach (/home/runner/PeskyBigheartedForms/node_modules/@rollup/plugin-typescript/dist/index.js:27:21)
    at Array.forEach (<anonymous>)
    at emitDiagnostics (/home/runner/PeskyBigheartedForms/node_modules/@rollup/plugin-typescript/dist/index.js:22:10)
    at Object.generateBundle (/home/runner/PeskyBigheartedForms/node_modules/@rollup/plugin-typescript/dist/index.js:474:13)
    at Promise.resolve.then (/home/runner/PeskyBigheartedForms/node_modules/rollup/dist/rollup.js:12899:25)
    at process._tickCallback (internal/process/next_tick.js:68:7)

exit status 1
```

**Comment by NotWoods** [comment]
You can tell the plugin to compile anyways by [setting `noEmitOnError` to false](https://github.com/rollup/plugins/tree/master/packages/typescript#noemitonerror).

**Comment by NotWoods** [comment]
If you explicitly set `lib`, does the error go away?

**Comment by NickGlenn** [comment]
Hey @NotWoods ,

Sadly, `noEmitOnError` doesn't force a build in this case and explicitly setting `lib` doesn't solve the issue either. I'm _guessing_ the cause is that `react-router-dom` brings in `@types/history`, which has a `History` interface that clashes with the `"dom"` library's `History` type. Did something change with how TS globals are handled since the previous version of this plugin?

**Comment by NotWoods** [comment]
Yes, the current version turns on typechecking as we previously did not do any checks beyond transpiling the file. If this compiles with Typescript normally the plugin should handle it, so I'll try to fix the bug.

**Comment by NickGlenn** [comment]
Awesome, thanks @NotWoods

**Comment by Paosder** [comment]
Is there any workaround or progress for this? I'm stuck in here.

**Comment by NickGlenn** [comment]
@Paosder Although not ideal, I've been using the original `rollup-plugin-typescript` module without issue. You may consider using that for the time being until this issue is resolved.

**Comment by NotWoods** [comment]
Agreed. The next plugin version will be published soon (and is already on master).

**Comment by NickGlenn** [comment]
@NotWoods This is exciting news, thanks to you and the team for working on this!

**Comment by shellscape** [comment]
@NotWoods need me to cut a new release? (sorry, work has been murder lately)

**Comment by NotWoods** [comment]
@shellscape Please do 😄 I sent you a list of updated plugins on Discord

**Comment by shellscape** [comment]
Done

**Comment by NotWoods** [comment]
Fixed in version 4.0.0 of the typescript plugin
